reactComponents/src/searchbar/tokenSearch/helpers/async.ts

```diff
@@ -136,7 +138,19 @@ export const searchTokensAsync = async (
     })
     // Flattening all the data sets into one data set.
     .flat()
-    .filter((pair: any) => pair.token0 && pair.token1)
+    .filter((pair: any) => {
+
+      // This checks if the pair's exchange and network is included in the network-exchange list
+      // passed to the searchbox component.
+      // This is a hacky fix. Ideally we should use the 'in' property in the Hasura search filters
+      // to only include valid exchanges per network pair search
```

Comment:
please read inline comments as this is just a band aid solution

Reply:
I agree. We can improve the logic that builds the query for romenet to do this. If we're going with this simple solution for now we need to create a tech debt ticket for the better solution.

Reply:
@oneway-fullstack